OTGTool/src/main/scala/t/platform/PlatformRegistry.scala

```diff
@@ -97,17 +110,17 @@ class PlatformRegistry(loader: PlatformLoader) {
 
   /**
    * Filter probes for a number of platforms.
+   * @param probes probes to be filtered (if empty, the result will contain all probes from the supplied platforms)
+   * @param platforms platforms that the result will be taken from
+   * @param species the preferred species of the return value, if any. For some platforms, this may not have any effect.
    */
-  def filterProbes(probes: Iterable[String],
-      platforms: Iterable[String],
-      species: Option[Species] = None): Iterable[String] = {
-    var rem = Set() ++ probes
-    var r = Set[String]()
-    for (p <- platforms; valid = filterProbes(rem, p, species)) {
-      rem --= valid
```

Comment:
This logic with a "remaining" (rem) set to be filtered, which is gradually subtracted from, is a legacy concept that is no longer needed here. Probes belong to exactly one platform, and I don't think there was much efficiency to be gained by complicating the method like this